commit 4eed7f5a8334a179f40b2c78c1ead572b9dd04a0
Author: LI Qingwu <Qing-wu.Li@leica-geosystems.com.cn>
Date:   Fri Mar 19 11:02:36 2021 +0000

    power: supply: bq27xxx: Add support for BQ78Z100
    
    Add support for TI BQ78Z100, I2C interface gas gauge.
    It provides a fully integrated safety protection
    and authentication for 1 to 2-series cell Li-Ion and
    Li-Polymer battery packs.
    
    The patch was tested with BQ78Z100 equipment.
    
    CASE I:  Discharging:
            POWER_SUPPLY_NAME=bq78z100-0
            POWER_SUPPLY_STATUS=Discharging
            POWER_SUPPLY_PRESENT=1
            POWER_SUPPLY_VOLTAGE_NOW=3386000
            POWER_SUPPLY_CURRENT_NOW=-5000
            POWER_SUPPLY_CAPACITY=27
            POWER_SUPPLY_CAPACITY_LEVEL=Normal
            POWER_SUPPLY_TEMP=269
            POWER_SUPPLY_TIME_TO_EMPTY_NOW=1249920
            POWER_SUPPLY_TECHNOLOGY=Li-ion
            POWER_SUPPLY_CHARGE_FULL=6494000
            POWER_SUPPLY_CHARGE_NOW=1736000
            POWER_SUPPLY_CHARGE_FULL_DESIGN=6000000
            POWER_SUPPLY_CYCLE_COUNT=1
            POWER_SUPPLY_POWER_AVG=-20000
            POWER_SUPPLY_HEALTH=Good
            POWER_SUPPLY_MANUFACTURER=Texas Instruments
    
    CASE II : No discharging current:
            POWER_SUPPLY_NAME=bq78z100-0
            POWER_SUPPLY_STATUS=Not charging
            POWER_SUPPLY_PRESENT=1
            POWER_SUPPLY_VOLTAGE_NOW=3386000
            POWER_SUPPLY_CURRENT_NOW=0
            POWER_SUPPLY_CAPACITY=27
            POWER_SUPPLY_CAPACITY_LEVEL=Normal
            POWER_SUPPLY_TEMP=270
            POWER_SUPPLY_TECHNOLOGY=Li-ion
            POWER_SUPPLY_CHARGE_FULL=6494000
            POWER_SUPPLY_CHARGE_NOW=1734000
            POWER_SUPPLY_CHARGE_FULL_DESIGN=6000000
            POWER_SUPPLY_CYCLE_COUNT=1
            POWER_SUPPLY_POWER_AVG=0
            POWER_SUPPLY_HEALTH=Good
            POWER_SUPPLY_MANUFACTURER=Texas Instruments
    
    Signed-off-by: LI Qingwu <Qing-wu.Li@leica-geosystems.com.cn>
    Reviewed-by: Krzysztof Kozlowski <krzk@kernel.org>
    Signed-off-by: Sebastian Reichel <sebastian.reichel@collabora.com>

diff --git a/drivers/power/supply/bq27xxx_battery_i2c.c b/drivers/power/supply/bq27xxx_battery_i2c.c
index eb4f4284982f..46f078350fd3 100644
--- a/drivers/power/supply/bq27xxx_battery_i2c.c
+++ b/drivers/power/supply/bq27xxx_battery_i2c.c
@@ -248,6 +248,7 @@ static const struct i2c_device_id bq27xxx_i2c_id_table[] = {
 	{ "bq27z561", BQ27Z561 },
 	{ "bq28z610", BQ28Z610 },
 	{ "bq34z100", BQ34Z100 },
+	{ "bq78z100", BQ78Z100 },
 	{},
 };
 MODULE_DEVICE_TABLE(i2c, bq27xxx_i2c_id_table);
@@ -284,6 +285,7 @@ static const struct of_device_id bq27xxx_battery_i2c_of_match_table[] = {
 	{ .compatible = "ti,bq27z561" },
 	{ .compatible = "ti,bq28z610" },
 	{ .compatible = "ti,bq34z100" },
+	{ .compatible = "ti,bq78z100" },
 	{},
 };
 MODULE_DEVICE_TABLE(of, bq27xxx_battery_i2c_of_match_table);